packages/sources/tradingeconomics/src/adapter.ts

```diff
@@ -1,36 +1,90 @@
-import { Requester, Validator, AdapterError } from '@chainlink/ea-bootstrap'
-import { ExecuteWithConfig, ExecuteFactory } from '@chainlink/types'
-import { Config, makeConfig, DEFAULT_ENDPOINT } from './config'
-import { markets } from './endpoint'
+import { Requester, Validator } from '@chainlink/ea-bootstrap'
+import { TEClient } from 'tradingeconomics-stream'
+import { AdapterRequest } from '@chainlink/types'
+import { Config, makeConfig } from './config'
 
-const inputParams = {
-  endpoint: false,
+let isServiceStarted = false
+const prices: { [symbol: string]: number } = {}
+
+const subscribe = (asset: string, config: Config) => {
+  const client: any = new TEClient({
+    ...config,
+    reconnect: true,
+  })
+
+  client.subscribe(asset)
+
+  client.on('message', (msg: any) => {
+    console.log(`Got price for asset ${asset}:`, msg.price)
+    prices[asset] = msg.price
+  })
 }
 
-export const execute: ExecuteWithConfig<Config> = async (request, config) => {
-  const validator = new Validator(request, inputParams)
-  if (validator.error) throw validator.error
+export const startService = (config: Config): void => {
+  const symbols = config.symbols.split(',')
+  for (let i = 0; i < symbols.length; i++) {
+    const symbol = commonSymbols[symbols[i]]
+    subscribe(symbol, config)
+  }
+}
 
-  Requester.logConfig(config)
+const customParams = {
+  base: ['base', 'from', 'asset'],
+}
+
+const commonSymbols: { [symbol: string]: string } = {
+  N225: 'NKY:IND',
+  FTSE: 'UKX:IND',
+}
+
+export const execute = async (input: AdapterRequest, config: Config) => {
+  // First start background service
+  if (!isServiceStarted) {
```

Comment:
@boxhock this PR is hard to review, but I essentially only changed the subscription flow by adding these few lines here and removing them from `index.ts`.

Reply:
I like the way of doing this :+1: